### PR TITLE
feat: add boostd-data metrics

### PR DESCRIPTION
**Pull request by LexLuthr**
Closes https://github.com/filecoin-project/boost/issues/1781

Pending tasks

- [x] Fix tests
- [x] Remove older dashboards
- [x] Fix boost-data dashboard
- [x] Explore if any other panels are required
- [x] Update grafana config to use a default dashboard
- [x] document all metrics and how can users scrape them
- [x] [update docker monitoring docs](https://app.gitbook.com/o/-L_E2woSLfhpBp2IPz9k/s/fZMqMjrWeFXMBES7ehiI/~/changes/143/monitoring)
- [x] ~~setup alertmanager~~ X Deferred for a later PR
- [x] Add permanent storage for tempo and prometheus

The updated documents will be merged only after this PR is merged as some of the referenced files do no exist in the current main branch.

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -84,9 +86,65 @@ var (
 	GraphsyncRequestBytesSentPaidCount          = stats.Int64("graphsync/request_bytes_sent_paid_count", "Counter of Graphsync paid bytes sent", stats.UnitBytes)
 	GraphsyncRequestBytesSentUnpaidCount        = stats.Int64("graphsync/request_bytes_sent_unpaid_count", "Counter of Graphsync unpaid bytes sent", stats.UnitBytes)
 	GraphsyncRequestNetworkErrorCount           = stats.Int64("graphsync/request_network_error_count", "Counter of Graphsync network errors", stats.UnitDimensionless)
+
+	//boostd-data
```
Comment (nonsense) [inline metrics/metrics.go:90]: `boostd-data` does not import `boostd` at all and it shouldn't.

**Comment by LexLuthr** [comment]
Pending tasks

- [ ] Fix tests
- [ ] Remove older dashboards
- [ ] Fix boost-data dashboard
- [ ] Find a way to add metrics correct and explore if any other panels are required
- [ ] Update grafana config to use a default dashboard
- [ ] document how to run grafana locally
- [ ] update docker monitoring docs
- [ ] setup alertmanager

File: cmd/booster-http/run.go [hunk 2]
```diff
@@ -171,8 +180,29 @@ var runCmd = &cli.Command{
 			}()
 		}
 
+		ctxx := lcli.ReqContext(cctx)
+		var ctx context.Context
```
Comment (nonsense) [inline cmd/booster-http/run.go:184]: Nit, but you don't need `ctxx` and you don't need `else statement`
```suggestion
		ctx := lcli.ReqContext(cctx)
```

File: extern/boostd-data/ldb/service.go [hunk 3]
```diff
@@ -47,6 +50,7 @@ type Store struct {
 	sync.Mutex
 	db       *DB
 	repopath string
+	ctx      context.Context
```
Comment (nonsense) [inline extern/boostd-data/ldb/service.go:53]: I don't understand why you need this global context?

File: extern/boostd-data/yugabyte/service.go [hunk 4]
```diff
@@ -145,26 +162,52 @@ func (s *Store) createPieceMetadata(ctx context.Context, pieceCid cid.Cid) error
...
+			stats.Record(s.ctx, metrics.SuccessGetOffsetSizeCount.M(1))
+		}
+	}()
 
 	var offset, size uint64
 	qry := `SELECT BlockOffset, BlockSize FROM PieceBlockOffsetSize WHERE PieceCid = ? AND PayloadMultihash = ?`
 	err := s.session.Query(qry, pieceCid.Bytes(), hash).WithContext(ctx).Scan(&offset, &size)
 	if err != nil {
 		err = normalizePieceCidError(pieceCid, err)
+		stats.Record(s.ctx, metrics.FailureGetOffsetSizeCount.M(1))
```
Comment (nonsense) [inline extern/boostd-data/yugabyte/service.go:189]: Why do we increment twice that metric -- once here and once in the defer statement?

Comment (nonsense) [inline extern/boostd-data/yugabyte/service.go:189]:
```suggestion
```

**Review by nonsense — APPROVED** [review]

**Pull request merged**